pypeit/spectrographs/mmt_bluechan.py

```diff
@@ -0,0 +1,296 @@
+            """
+            date = headarr[0]['DATE-OBS']
+            ut = headarr[0]['UT']
+            ttime = Time(f"{date}T{ut}", format='isot')
+            return ttime.mjd
+        elif meta_key == 'lampstat01':
+            """
+            If the comparison mirror is in, there will be a 'COMPLAMP' header entry containing the lamps
+            that are turned on. However, if the comparison mirror is out, then this header entry doesn't exist.
+            So need to test for it and set to 'Off' if it's not there.
+            """
+            if 'COMPLAMP' in headarr[0]:
+                return headarr[0]['COMPLAMP']
+            else:
+                return 'off'
+        else:
+            msgs.error(f"Not ready for compound meta, {meta_key}, for MMT Blue Channel.")
+
+
+    def default_pypeit_par(self):
+        """
+        Set default parameters for MMT/Blue Channel reductions.
+        """
+        par = pypeitpar.PypeItPar()
+        par['rdx']['spectrograph'] = 'mmt_bluechannel'
+        # Wavelengths
+        # 1D wavelength solution
+        par['calibrations']['wavelengths']['rms_threshold'] = 0.5
+        par['calibrations']['wavelengths']['sigdetect'] = 5.
+        par['calibrations']['wavelengths']['fwhm']= 5.0
+        # HeNeAr is by far most commonly used, though ThAr is used for some situations.
+        par['calibrations']['wavelengths']['lamps'] = ['ArI', 'ArII', 'HeI', 'NeI']
+        par['calibrations']['wavelengths']['method'] = 'holy-grail'
+
+        # Tilt and slit parameters
+        par['calibrations']['tilts']['tracethresh'] =  10.0
+        par['calibrations']['tilts']['spat_order'] = 6
+        par['calibrations']['tilts']['spec_order'] = 6
+        par['calibrations']['slitedges']['sync_predict'] = 'nearest'
+
+        # Processing steps
+        turn_off = dict(use_biasimage=False, use_darkimage=False)
+        par.reset_all_processimages_par(**turn_off)
+
+        # Extraction
+        par['reduce']['skysub']['bspline_spacing'] = 0.8
+        par['reduce']['extraction']['sn_gauss'] = 4.0
+        ## Do not perform global sky subtraction for standard stars
+        par['reduce']['skysub']['global_sky_std']  = False
+
+        # Flexure
+        par['flexure']['spec_method'] = 'boxcar'
+
+        # cosmic ray rejection parameters for science frames
+        par['scienceframe']['process']['sigclip'] = 5.0
+        par['scienceframe']['process']['objlim'] = 2.0
+
+        # Set the default exposure time ranges for the frame typing
+        par['calibrations']['standardframe']['exprng'] = [None, 600]
+        par['calibrations']['arcframe']['exprng'] = [10, None]
+        par['calibrations']['darkframe']['exprng'] = [300, None]
+        par['scienceframe']['exprng'] = [10, None]
+
+        # Sensitivity function parameters
+        par['sensfunc']['polyorder'] = 7
+        par['sensfunc']['IR']['telgridfile'] = resource_filename('pypeit', '/data/telluric/TelFit_Paranal_VIS_4900_11100_R25000.fits')
```

Comment:
Is Paranal the best match to MMTO?
Not sure..

Reply:
well, it's closer than mauna kea at any rate, but blue channel is only usable < 0.9 um, anyway, so it doesn't matter too much, i think. a more appropriate model would be https://mwvgroup.github.io/pwv_kpno.

Reply:
agreed.